api/app/controllers/gift_cards_controller.rb

```diff
@@ -1,8 +1,63 @@
+require 'stripe'
+
 class GiftCardsController < ApplicationController
+  before_action :set_gift_card, only: [:show, :update, :destroy]
 
   # POST /gift_cards
   def create
-    obj = { message: 'hello world!' }
-    json_response(obj)
+    @gift_card = GiftCard.create!(create_params)
+    json_response(@gift_card, :created)
+  end
+
+  # GET /gift_cards/:d
+  def show
+    json_response(@gift_card)
+  end
+
+  # PUT /gift_cards/:id
+  def update
+    # only allows updating amounts of gift cards
+    original_amount = @gift_card.amount
+    new_amount = update_params[:amount].to_i
+
+    if original_amount < new_amount
+      json_response({ message: 'Cannot increase gift card amount' }, :unprocessable_entity)
```

Comment:
Could you raise an exception instead? The exception handler should catch it. See charges controller as an example.